### PR TITLE
use ark to download & install from tarball

### PR DESCRIPTION
**Pull request by mburns**
This PR is an attempt to use [ark](https://github.com/burtlo/ark) instead of manually downloading and extracting the tarball.

### CODE REVIEW

File: spec/unit/recipes/default_spec.rb [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+
+  it 'includes the apt recipe' do
+    expect(chef_run).to include_recipe('apt')
+  end
 
   it 'includes the java recipe' do
     expect(chef_run).to include_recipe('java')
   end
 
+  it 'renders the downloaded tarball' do
```
Comment (dwradcliffe) [inline spec/unit/recipes/default_spec.rb]: Description could be better here

File: spec/unit/recipes/default_spec.rb [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
+    expect(chef_run).to install_ark('solr')
+  end
+
+  it 'renders the data directory' do
```
Comment (dwradcliffe) [inline spec/unit/recipes/default_spec.rb]: Description could be better here too

**Comment by dwradcliffe** [comment]
Couple small things with the tests. Also needs a rebase/squash I think.

**Comment by mburns** [comment]
ping

**Comment by mburns** [comment]
Sorry about that, @eric-tucker. You can version lock to `0.5.0` for the time being. I agree, the ideal behavior is to not download a tarball if it already exists with a known checksum. Let me take a crack at fixing this next week.

**Comment by eric-tucker** [comment]
@mburns we went to version 0.6.0 for #35 I had put in.  It seems that @dwradcliffe slipped this in at the same time which we didn't intend.

We worked around the issue by setting the value of "url" to a file:// value and downloading in advance so that ark doesn't go out to the internet.  That allows us to continue to download our solr tarball from our private S3 location in advance using s3_file and an IAM role and continue to use this cookbook.

For now, it's probably safe to leave this as is.  I'd rather not risk breaking our work-around again.

**Pull request merged**